### PR TITLE
More descriptive error for a bad ref in repos.git

### LINKED ISSUE #771: Misleading error when using repos.git
If you use a non-existent reference in the repos.git the error returned is that it could not clone the repo. The real error is that it could not archive that reference/tag, so the error needs to be improved to make it easier to debug and fix the `ref` entry in the blueprint.

### PR DESCRIPTION
**Pull request by evan-goode**
Fixes #771

I also redirected stderr to stdout in a couple subprocess calls so that git's error messages are captured in the log.

### CODE REVIEW

**Comment by atodorov** [comment]
Can one of the admins verify this patch?

**Comment by bcl** [comment]
Looks good so far, could you also add a test for the archive failure case?

**Comment by evan-goode** [comment]
I may be misunderstanding, but it looks like there already is one:
https://github.com/weldr/lorax/blob/c1aa57022bdc99b8609ae3fc9296302d6420c626/tests/pylorax/test_gitrpm.py#L118

**Comment by bcl** [comment]
> I may be misunderstanding, but it looks like there already is one:

Oh, right, we're using RuntimeError for both. I think it would be good to test for the two failure strings 'Failed to clone' and 'Failed to archive' just to be sure we're telling the users something useful.

File: tests/pylorax/test_gitrpm.py [hunk 1]
```diff
@@ -111,9 +111,10 @@ def git_fail_repo_test(self):
             ref="v1.1.0"
             destination="/srv/testing-rpm/"
         """ % ("/tmp/no-repo-here/"))
-        with self.assertRaises(RuntimeError):
+        with self.assertRaises(RuntimeError) as cm:
```
Comment (bcl) [inline tests/pylorax/test_gitrpm.py]: Small nitpick, sorry, we use 'as e' for error results, not 'cm'. Looks good other than that.

Comment (evan-goode) [reply tests/pylorax/test_gitrpm.py]: Sure, `e` works.

**Comment by chris1984** [comment]
ok to test

**Comment by chris1984** [comment]
[test]

**Comment by chris1984** [comment]
test

**Comment by bcl** [comment]
Merging this. Tried to force new tests with #778 but infrastructure failed. Version pulled into rhel8-branch ran fine, also passes locally.

**Pull request merged**